GoogleForCreators/web-stories-wp
# Add Text Style Presets

## Summary
- Adds a separate style presets panel
- Reuses the same logic for the colors vs style presets

Note that this does NOT include style changes for the Color Presets nor general style changes --- the Remove icon used in both preset panels is still the same and could be changed within #2908 for everything together if need be.


## Relevant Technical Choices



## To-do

- [x] Add style presets functionality, including all inline styles
- [x] Adjust existing unit tests
- [x] Karma tests
- [x] Transparent bg style -- needs UX confirmation (DONE)



## User-facing changes



## Testing Instructions
I
1. Add a Text element
2. Verify that Saved styles panel is visible.
3. Add a Media element and select both elements
4. Verify that the Saved Styles panel is not visible.

II
1. Add a Text element and change its style (for the whole text): font, font-weight, letter-spacing, italic, bold, background color/highlight, line-height, underline, text color, padding.
2. Click on the `+` icon next to the "Saved styles" to add a text preset.
3. Verify that a preset is added accordingly.
4. Add a new text now.
5. Click on the preset while the text is selected.
6. Verify that the same styles are now applied on the new text element.
7. Add more text elements and verify that the style is added on multi-selection, too

III
1. Add a few text elements and change their styles separately.
2. Multi-select the added elements and click the icon to add the presets again.
3. Verify that multiple style presets are added at the same time, one per each text.

IV
1. Add a text element with some inline styles applied on part of the text only.
2. Now add a preset.
3. Verify that the styles that are applied partially on the text only are not considered in the saved preset.

V
1. Add a few text elements and create some presets.
2. Click the Edit icon for the "Saved styles"
3. Verify that the edit mode is entered with remove icons appearing on the presets.
4. Click on the Remove icon and verify the presets being removed.


---



Fixes #353
